// @dasha.ai/sdk/dist/zip.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.application = exports.jszipFiles = exports.file = exports.directory = void 0;
const log_1 = require("./log");
const zip_helpers_1 = require("./zip-helpers");
const errors = __importStar(require("./errors"));
// the code here is written as if `archiver` was an optional dependency
// based on the assumption that we may want to work with pre-packed applications
// in the future
/** @internal */
async function directory(path, log) {
    const { default: archiver } = await Promise.resolve().then(() => __importStar(require("archiver"))).catch(() => {
        log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("archiver"));
    });
    return new Promise((resolve, reject) => {
        const zip = archiver("zip");
        const chunks = [];
        zip.on("error", (error) => reject(log_1.logAndReturn(log, new errors.FileSystemError(error))));
        zip.on("warning", (error) => reject(log_1.logAndReturn(log, new errors.FileSystemError(error))));
        zip.on("data", (chunk) => chunks.push(chunk));
        zip.on("finish", () => resolve(Buffer.concat(chunks)));
        zip.directory(path, false);
        zip.finalize().catch((error) => reject(log_1.logAndReturn(log, new errors.FileSystemError(error))));
    });
}
exports.directory = directory;
/** @internal */
async function jszipFile(name, content, log) {
    const { default: JSZip } = await Promise.resolve().then(() => __importStar(require("jszip")));
    const zip = new JSZip();
    zip.file(name, content);
    const array = await zip.generateAsync({ type: "uint8array" }).catch((error) => {
        log_1.logAndThrow(log, new errors.FileSystemError(error));
    });
    return array;
}
/** @internal */
async function archiverFile(name, value, log) {
    const { default: archiver } = await Promise.resolve().then(() => __importStar(require("archiver")));
    const stream = await Promise.resolve().then(() => __importStar(require("stream")));
    return new Promise((resolve, reject) => {
        const zip = archiver("zip");
        const chunks = [];
        zip.on("error", (error) => reject(log_1.logAndReturn(log, new errors.FileSystemError(error))));
        zip.on("warning", (error) => reject(log_1.logAndReturn(log, new errors.FileSystemError(error))));
        zip.on("data", (chunk) => chunks.push(chunk));
        zip.on("finish", () => resolve(Buffer.concat(chunks)));
        zip.append(value, { name });
        zip.finalize().catch((error) => reject(log_1.logAndReturn(log, new errors.FileSystemError(error))));
    });
}
/** @internal */
async function file(name, content, log) {
    try {
        await Promise.resolve().then(() => __importStar(require("archiver"))).catch(() => {
            log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("archiver"));
        });
        return await archiverFile(name, content, log);
    }
    catch (e) {
        await Promise.resolve().then(() => __importStar(require("jszip"))).catch(() => {
            log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("jszip"));
        });
        return await jszipFile(name, content, log);
    }
}
exports.file = file;
/** @internal */
async function jszipFiles(zipData, log) {
    const { default: JSZip } = await Promise.resolve().then(() => __importStar(require("jszip")));
    const zip = new JSZip();
    for (const [path, content] of Object.entries(zipData)) {
        zip.file(path, content);
    }
    return await zip
        .generateAsync({ type: "uint8array" })
        .catch((error) => log_1.logAndThrow(log, new errors.FileSystemError(error)));
}
exports.jszipFiles = jszipFiles;
/** @internal */
async function archiverFiles(zipData, log) {
    const { default: archiver } = await Promise.resolve().then(() => __importStar(require("archiver")));
    return new Promise((resolve, reject) => {
        const zip = archiver("zip");
        const chunks = [];
        zip.on("error", (error) => reject(error));
        zip.on("warning", (error) => reject(error));
        zip.on("data", (chunk) => chunks.push(chunk));
        zip.on("finish", () => resolve(Buffer.concat(chunks)));
        for (const [name, content] of Object.entries(zipData))
            zip.append(content, { name });
        zip.finalize().catch((error) => reject(log_1.logAndReturn(log, new errors.FileSystemError(error))));
    });
}
/** @internal */
async function application(appFolderPath, log) {
    await Promise.resolve().then(() => __importStar(require("fs"))).catch(() => {
        log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("fs"));
    });
    await Promise.resolve().then(() => __importStar(require("glob"))).catch(() => {
        log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("glob"));
    });
    await Promise.resolve().then(() => __importStar(require("path"))).catch(() => {
        log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("path"));
    });
    const zipData = await zip_helpers_1.prepareAppZipData(appFolderPath);
    try {
        await Promise.resolve().then(() => __importStar(require("archiver"))).catch(() => {
            log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("archiver"));
        });
        return await archiverFiles(zipData, log).catch((e) => log_1.logAndThrow(log, new errors.FileSystemError(e)));
    }
    catch (e) {
        await Promise.resolve().then(() => __importStar(require("jszip"))).catch(() => {
            log_1.logAndThrow(log, new errors.OptionalDependencyNotFoundError("jszip"));
        });
        return await jszipFiles(zipData, log).catch((e) => log_1.logAndThrow(log, new errors.FileSystemError(e)));
    }
}
exports.application = application;
