// @dasha.ai/sdk/dist/rest-api/generated/core/api/paraphrase-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParaphraseApi = exports.ParaphraseApiFactory = exports.ParaphraseApiFp = exports.ParaphraseApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ParaphraseApi - axios parameter creator
 * @export
 */
const ParaphraseApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {number} [temperature]
         * @param {number} [paraphraseMaxNumber]
         * @param {number} [adequacyThreshold]
         * @param {number} [fluencyThreshold]
         * @param {DiversityRankerType} [diversityRanker]
         * @param {Array<string>} [requestBody]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ParaphraseBatchPost: async (temperature, paraphraseMaxNumber, adequacyThreshold, fluencyThreshold, diversityRanker, requestBody, options = {}) => {
            const localVarPath = `/api/v1/paraphrase/batch`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            if (temperature !== undefined) {
                localVarQueryParameter['Temperature'] = temperature;
            }
            if (paraphraseMaxNumber !== undefined) {
                localVarQueryParameter['ParaphraseMaxNumber'] = paraphraseMaxNumber;
            }
            if (adequacyThreshold !== undefined) {
                localVarQueryParameter['AdequacyThreshold'] = adequacyThreshold;
            }
            if (fluencyThreshold !== undefined) {
                localVarQueryParameter['FluencyThreshold'] = fluencyThreshold;
            }
            if (diversityRanker !== undefined) {
                localVarQueryParameter['DiversityRanker'] = diversityRanker;
            }
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(requestBody, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {number} [paraphraseMaxNumber]
         * @param {number} [temperature]
         * @param {string} [body]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ParaphraseGptPost: async (paraphraseMaxNumber, temperature, body, options = {}) => {
            const localVarPath = `/api/v1/paraphrase/gpt`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            if (paraphraseMaxNumber !== undefined) {
                localVarQueryParameter['ParaphraseMaxNumber'] = paraphraseMaxNumber;
            }
            if (temperature !== undefined) {
                localVarQueryParameter['Temperature'] = temperature;
            }
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(body, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ParaphraseApiAxiosParamCreator = ParaphraseApiAxiosParamCreator;
/**
 * ParaphraseApi - functional programming interface
 * @export
 */
const ParaphraseApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.ParaphraseApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @param {number} [temperature]
         * @param {number} [paraphraseMaxNumber]
         * @param {number} [adequacyThreshold]
         * @param {number} [fluencyThreshold]
         * @param {DiversityRankerType} [diversityRanker]
         * @param {Array<string>} [requestBody]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ParaphraseBatchPost(temperature, paraphraseMaxNumber, adequacyThreshold, fluencyThreshold, diversityRanker, requestBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ParaphraseBatchPost(temperature, paraphraseMaxNumber, adequacyThreshold, fluencyThreshold, diversityRanker, requestBody, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @param {number} [paraphraseMaxNumber]
         * @param {number} [temperature]
         * @param {string} [body]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ParaphraseGptPost(paraphraseMaxNumber, temperature, body, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ParaphraseGptPost(paraphraseMaxNumber, temperature, body, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.ParaphraseApiFp = ParaphraseApiFp;
/**
 * ParaphraseApi - factory interface
 * @export
 */
const ParaphraseApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.ParaphraseApiFp(configuration);
    return {
        /**
         *
         * @param {number} [temperature]
         * @param {number} [paraphraseMaxNumber]
         * @param {number} [adequacyThreshold]
         * @param {number} [fluencyThreshold]
         * @param {DiversityRankerType} [diversityRanker]
         * @param {Array<string>} [requestBody]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ParaphraseBatchPost(temperature, paraphraseMaxNumber, adequacyThreshold, fluencyThreshold, diversityRanker, requestBody, options) {
            return localVarFp.apiV1ParaphraseBatchPost(temperature, paraphraseMaxNumber, adequacyThreshold, fluencyThreshold, diversityRanker, requestBody, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {number} [paraphraseMaxNumber]
         * @param {number} [temperature]
         * @param {string} [body]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ParaphraseGptPost(paraphraseMaxNumber, temperature, body, options) {
            return localVarFp.apiV1ParaphraseGptPost(paraphraseMaxNumber, temperature, body, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ParaphraseApiFactory = ParaphraseApiFactory;
/**
 * ParaphraseApi - object-oriented interface
 * @export
 * @class ParaphraseApi
 * @extends {BaseAPI}
 */
class ParaphraseApi extends base_1.BaseAPI {
    /**
     *
     * @param {number} [temperature]
     * @param {number} [paraphraseMaxNumber]
     * @param {number} [adequacyThreshold]
     * @param {number} [fluencyThreshold]
     * @param {DiversityRankerType} [diversityRanker]
     * @param {Array<string>} [requestBody]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ParaphraseApi
     */
    apiV1ParaphraseBatchPost(temperature, paraphraseMaxNumber, adequacyThreshold, fluencyThreshold, diversityRanker, requestBody, options) {
        return exports.ParaphraseApiFp(this.configuration).apiV1ParaphraseBatchPost(temperature, paraphraseMaxNumber, adequacyThreshold, fluencyThreshold, diversityRanker, requestBody, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {number} [paraphraseMaxNumber]
     * @param {number} [temperature]
     * @param {string} [body]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ParaphraseApi
     */
    apiV1ParaphraseGptPost(paraphraseMaxNumber, temperature, body, options) {
        return exports.ParaphraseApiFp(this.configuration).apiV1ParaphraseGptPost(paraphraseMaxNumber, temperature, body, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ParaphraseApi = ParaphraseApi;
